src/Validator.ts

```diff
@@ -1,23 +1,32 @@
+export enum ErrorType {
+  InputData = 0,
+  FileFormat = 1,
+  ChannelCount = 2,
+  FseqType = 3,
+  Duration = 4,
+  Memory = 5,
+}
+
 /**
  * Validation result type
- * Validation result is considered valid if error field is undefined
- * If the error field is specified, an error message will be built and the validation should be considered invalid
+ * Validation result is considered valid if the errors list is undefined, otherwise invalid.
  * Other fields, if available, are returned to aid in building detailed errors or descriptions
  * @typedef {Object} ValidationResults - creates a new type named 'ValidationResults'
  * @property {number} [frameCount=0] - number of frames
  * @property {number} [memoryUsage=0] - memory usage, ranging from 0 to 1 (1 = 100%), commandCount / 681 = memoryUsage
  * @property {number} [duration=0] - duration in milliseconds
  * @property {number} [commandCount=0] - number of commands, maximum being 681. commandCount / 681 = memoryUsage
  * @property {number} [stepTime=0] - duration between frames
- * @property {(string | undefined)} [error=undefined] - error message, validation invalid if defined, valid if undefined
+ * @property {ErrorType[] | undefined} [errors=undefined] - If any, the errors that were found. If it undefined, the validation result is valid.
  */
-export type ValidationResults = {
+ export type ValidationResults = {
   frameCount: number;
   memoryUsage: number;
   duration: number;
   commandCount: number;
   stepTime: number;
-  error: string | undefined;
+  channelCount: number;
+  errors: ErrorType[] | undefined;
```

Comment:
I think `undefined` is unnecessary here. By default it can be an empty array (as you have it defined on line 45)